src/test/integration/suite/test-utils.ts

```diff
@@ -0,0 +1,38 @@
+import retry from "async-retry";
+import { EnvironmentModel } from "../../../get-environments";
+import * as vscode from "vscode";
+import sinon from "sinon";
+import { Credentials } from "../../../types";
+
+export const waitFor = <T>(callback: (...args: any[]) => T): Promise<T> => {
+  return retry(callback, { retries: 5, minTimeout: 300, maxTimeout: 1000 });
+};
+
+export const login = async (auth: Credentials) => {
+  const inputStub = sinon.stub(vscode.window, "showInputBox");
```

Comment:
Since there is no api in vsocde that will let me type into input box i need to mock it